code/game/objects/items/gamemode/survivor.dm

```diff
@@ -0,0 +1,340 @@
+/obj/item/laptop/rescue
+    name = "Communications laptop"
+    desc = "A device used for augmenting communication. Someone seems to have left it logged in with open communications to a nearby ship. Set this up next to a rescue beacon."
+    icon = 'icons/obj/machines/laptop_beacon.dmi'
+    icon_state = "tandy0"
+    luminosity = 4
+    resistance_flags = UNACIDABLE|INDESTRUCTIBLE
+    w_class = WEIGHT_CLASS_BULKY
+    var/activation_time = 5 SECONDS
+
+/obj/item/laptop/rescue/Destroy()
+    return ..()
+
+/obj/item/laptop/rescue/update_icon()
+    icon_state = "tandy0"
+    cut_overlays()
+    if(anchored)
+        icon_state = "tandy1"
+        add_overlay("tandy1o_wifi")
+
+    return ..()
+
+/obj/item/laptop/rescue/attack_hand(mob/user as mob)
+    if(!anchored)
+        return ..()
+    if(!ishuman(user)) 
+        return ..()
+
+    user.visible_message("<span class='notice'>[user] starts packing up \the [src].</span>",
+    "<span class='notice'>You start packing up \the [src]. <b>This will interrupt the process.</b></span>")
+    if(!user.get_active_held_item() && do_after(user, activation_time*1.2, TRUE, 5, BUSY_ICON_FRIENDLY,, TRUE))
+        reset_state()
+        user.put_in_hands(src)
+    . = ..()
+    
+/obj/item/laptop/rescue/attack_self(mob/user)
+    if(anchored)
+        to_chat(user, "<span class='warning'>It's already been anchored. Just leave it.</span>")
+        return
+
+    if(!ishuman(user)) 
+        return
+
+    if(!user.mind)
+        to_chat(user, "<span class='warning'>It doesn't seem to do anything for you.</span>")
+        return
+
+    var/area/A = get_area(user)
+    if(A && istype(A) && A.ceiling >= CEILING_METAL)
+        to_chat(user, "<span class='warning'>You have to be outside or under a glass ceiling to activate this.</span>")
+        return
+
+    user.visible_message("<span class='notice'>[user] starts setting up [src] on the ground.</span>",
+    "<span class='notice'>You start setting up [src] on the ground and inputting all the data it needs.</span>")
+
+    if(do_after(user, activation_time, TRUE, 5, BUSY_ICON_FRIENDLY,, TRUE))
```

Comment:
Early reteurn

Reply:
Generally i agree but that feels against the spirit of a `do_after` which make sense to have their "action" code indented.

Reply:
I wouldn't consider do_after as some sort of control keyword. If it makes sense to early return with one then do so and don't be restricted by what's common.